devenv/fetch.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+import argparse
+import os
+import sys
+from collections.abc import Sequence
+from typing import TypeAlias
+
+from devenv.constants import CI
+from devenv.constants import DARWIN
+from devenv.constants import EXTERNAL_CONTRIBUTOR
+from devenv.constants import homebrew_bin
+from devenv.lib import proc
+
+help = "Fetches a respository"
+ExitCode: TypeAlias = "str | int | None"
+
+
+def _fetch(
+    coderoot: str, repo: str, noauth: bool = False, sync: bool = True
```

Comment:
nit but i'd change `noauth=False` to `auth=False` so it's more obvious to callers that a repository is private

Reply:
Yeah, yeah.. I should avoid negatives in variable names.. Good point.